### PR TITLE
Refactor Category Controller Tests

### LINKED ISSUE #427: Refactor Category Controller Tests
# Problem

Adding api_case.ex helpers to Category controller tests.

## References

Progress on: #413

### PR DESCRIPTION
**Pull request by amyschools**
# What's in this PR?

Add api_case.ex helper functions to Category controller tests.

## References
Closes #427 

Progress on: #413

### CODE REVIEW

File: test/controllers/category_controller_test.exs [hunk 1]
```diff
@@ -1,102 +1,79 @@
...
-
-  def request_update(conn, attrs) do
-    category = insert(:category)
-    payload = json_payload(:category, attrs)
-    path = conn |> category_path(:update, category)
-
-    conn |> put(path, payload)
-  end
-
   test "lists all entries on index", %{conn: conn} do
```
Comment (mackenziehicks) [inline test/controllers/category_controller_test.exs]: looks like this is missing `describe "index" do`

File: test/controllers/category_controller_test.exs [hunk 2]
```diff
@@ -1,102 +1,79 @@
...
 
   describe "update" do
     @tag authenticated: :admin
     test "updates and renders chosen resource when data is valid", %{conn: conn} do
-      response = conn |> request_update(@valid_attrs) |> json_response(200)
-
-      category = Repo.get_by(Category, @valid_attrs)
-      assert category
-      assert response["data"]["id"] == "#{category.id}"
+      assert conn |> request_update(@valid_attrs) |> json_response(200)
```
Comment (mackenziehicks) [inline test/controllers/category_controller_test.exs:58]: For the organization membership update test:

`organization = insert(:organization)`
      `membership = insert(:organization_membership, organization: organization, role: "pending")`
      `insert(:organization_membership, organization: organization, member: current_user, role: "owner")`

```
  `assert conn |> request_update(membership, @valid_attrs) |> json_response(200)`
```

Should I change it to look more like this one?  Looks like the request_update helper has a default_record.

Comment (joshsmith) [reply test/controllers/category_controller_test.exs:58]: I don't think so, because you do need someone who's an organization admin to approve, right?

**Review by joshsmith — APPROVED** [review]

**Comment by joshsmith** [comment]
@amyschools that commit message is a little unclear.

**Pull request merged**